src/routes/choice-exercises/__tests__/ArticleChoiceExerciseScreen.spec.tsx

```diff
@@ -97,11 +97,13 @@ describe('ArticleChoiceExerciseScreen', () => {
     fireEvent(getByText('Der'), 'pressOut')
     fireEvent.press(getByText(labels.exercises.next))
     fireEvent(getByText('Das'), 'pressOut')
-    await fireEvent.press(getByText(labels.exercises.showResults))
 
-    await expect(saveExerciseProgress).toHaveBeenCalledWith(1, ExerciseKeys.articleChoiceExercise, [
+    await act(async () => {
+      await fireEvent.press(getByText(labels.exercises.showResults))
+    })
```

Comment:
act fix: wrap all functions in act, which lead to state updates. Here the press on showResults calls onExerciseFinished, which then calls initializeExercise, which then calls setCurrentWord and setResults -> state update